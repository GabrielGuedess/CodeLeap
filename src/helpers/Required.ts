export type Required<T> = {
  [P in keyof T]-?: T[P];
};
