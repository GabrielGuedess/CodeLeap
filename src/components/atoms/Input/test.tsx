import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import { Input } from '.';

describe('<Input />', () => {
  it('should render the input correctly', () => {
    // Arrange
    const { container } = renderWithTheme(<Input title="Title" />);

    // Assert
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the input with error', () => {
    renderWithTheme(<Input title="E-mail" errorMessage="email is invalid" />);

    // Assert
    expect(screen.getByText('email is invalid')).toBeInTheDocument();
  });
});
