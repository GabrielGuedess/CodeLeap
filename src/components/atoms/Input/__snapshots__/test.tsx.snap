// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render the input correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.8rem;
}

.c1 {
  font-size: 1.6rem;
}

.c2 {
  width: 100%;
  padding: 0.8rem 1.12rem;
  font-size: 1.4rem;
  border: 1px solid #777777;
  border-radius: 0.8rem;
}

.c2:focus {
  box-shadow: #B8BEF3 0px 0px 0px 1.5px;
}

.c2::-webkit-input-placeholder {
  color: #CCCCCC;
}

.c2::-moz-placeholder {
  color: #CCCCCC;
}

.c2:-ms-input-placeholder {
  color: #CCCCCC;
}

.c2::placeholder {
  color: #CCCCCC;
}

<div
  class="c0"
>
  <label
    class="c1"
    for=":r0:"
  >
    Title
  </label>
  <input
    class="c2"
    id=":r0:"
  />
</div>
`;
